src/main/java/org/takes/misc/Concat.java

```diff
@@ -0,0 +1,118 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.takes.misc;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * Concat iterable.
+ *
+ * @author Jason Wong (super132j@yahoo.com)
+ * @version $Id$
+ * @since 0.13.8
+ */
+public final class Concat<T> implements Iterable<T> {
+
+    /**
+     * Internal reference to hold the first elements from constructor.
+     */
+    private final transient Iterable<T> left;
+
+    /**
+     * Internal reference to hold the second elements from constructor.
+     */
+    private final transient Iterable<T> right;
+
+    /**
+     * To produce an iterable collection combining a and b, with order of the
+     * elements in a first.
+     * @param aitb First iterable to concat
+     * @param bitb Second iterable to conat
+     */
+    public Concat(final Iterable<T> aitb, final Iterable<T> bitb) {
+        this.left = aitb;
+        this.right = bitb;
+    }
+
+    @Override
+    public Iterator<T> iterator() {
+        return new ConcatIterator<T>(
+            this.left.iterator(),
+            this.right.iterator()
+        );
+    }
+
+    /**
+     * The concat iterator to traverse the input iterables as if they are
+     * from one list.
+     */
+    private static class ConcatIterator<E> implements Iterator<E> {
+
+        /**
+         * Internal reference for holding the first iterator form constructor.
+         */
+        private final transient Iterator<E> left;
+
+        /**
+         * Internal reference for holding the second iterator form constructor.
+         */
+        private final transient Iterator<E> right;
+
+        /**
+         * Ctor. ConcatIterator traverses the element.
+         * @param aitr The first iterable to traverse
+         * @param bitr The second iterable to traverse
+         */
+        public ConcatIterator(final Iterator<E> aitr, final Iterator<E> bitr) {
+            this.left = aitr;
+            this.right = bitr;
+        }
+
+        @Override
+        public boolean hasNext() {
+            final boolean first = this.left.hasNext();
+            final boolean second = this.right.hasNext();
+            return first || second;
+        }
+
+        @Override
+        public E next() {
+            final E object;
+            if (this.left.hasNext()) {
+                object = this.left.next();
+            } else if (this.right.hasNext()) {
+                object = this.right.next();
+            } else {
+                throw new NoSuchElementException();
+            }
+            return object;
+        }
+
+        @Override
+        public void remove() {
+            throw new UnsupportedOperationException();
```

Comment:
I think we should add some text here, to explain why we're rejecting `remove()` call. maybe something like `"this iterable is immutable, can't remove anything"`